core/src/main/java/org/apache/accumulo/core/iterators/SortedKeyValueIterator.java

```diff
@@ -147,4 +147,13 @@
    *              if not supported.
    */
   SortedKeyValueIterator<K,V> deepCopy(IteratorEnvironment env);
+
+  /**
+   * Closes the Iterator. This should be overridden by the implementing class that has access to <tt>SortedKeyValueIterator source</tt> provided in the
```

Comment:
I think Dylan addressed many of my concerns and perhaps a design doc would help me understand but why do we need to put state into the iterator stack? Why the need to make the stack closeable? What does this solve? 

It seems like we're putting state into the stack where state isn't necessary. The ticket says it would be useful, but wouldn't it be more useful to always ensure the data source itself is closed. Is the utility here to ensure that we give iterators the opportunity to break the previous paradigm of iteration and maintain state that shouldn't be maintained? 

TL;DR
Does the potential complication empower users or maintainers more than it currently does, and if so is there a specific use case that would drive a design?